sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py

```diff
@@ -650,44 +671,53 @@ def _quick_query_options(self, query_expression,
-        # type: (str, **Any) -> QuickQueryReader
+    def query_blob(self, query_expression, **kwargs):
+        # type: (str, **Any) -> BlobQueryReader
         """Enables users to select/project on blob/or blob snapshot data by providing simple query expressions.
-        This operations returns a QuickQueryReader, users need to use readall() or readinto() to get query data.
+        This operations returns a BlobQueryReader, users need to use readall() or readinto() to get query data.
 
         :param str query_expression:
             Required. a query statement.
-        :keyword func(~azure.storage.blob.QuickQueryError, int, int) progress_callback:
-            Callback where the caller can track progress of the operation as well as the quick query failures.
-        :keyword input_serialization:
-            Optional. Defines the input serialization for a blob quick query request.
-            This keyword arg could be set for delimited (CSV) serialization or JSON serialization.
-            When the input_serialization is set for JSON records, only a record separator in str format is needed.
-        :paramtype input_serialization: ~azure.storage.blob.DelimitedTextConfiguration or str
-        :keyword output_serialization:
-            Optional. Defines the output serialization for a blob quick query request.
-            This keyword arg could be set for delimited (CSV) serialization or JSON serialization.
-            When the input_serialization is set for JSON records, only a record separator in str format is needed.
-        :paramtype output_serialization: ~azure.storage.blob.DelimitedTextConfiguration or str.
+        :keyword Union[str, Callable[Exception]] errors:
+            Determines the error behaviour. The default value is 'strict', where any non-fatal error will be
+            raised. Other possible values include:
+                - 'ignore': Non-fatal errors will be ignored, this may result in dropped records.
+                - Callable[Exception]: If a callable is provided, customer error handling can be defined.
+        :keyword blob_format:
+            Optional. Defines the serialization of the data currently stored in the blob. The default is to
+            treat the blob data as CSV data formatted in the default dialect. This can be overridden with
+            a custom CSVDialect, or alternatively a DelimitedJSON.
+        :paramtype blob_format: ~azure.storage.blob.CSVDialect or ~azure.storage.blob.DelimitedJSON
+        :keyword output_format:
+            Optional. Defines the output serialization for the data stream. By default the data will be returned
+            as it is represented in the blob. By providing an output format, the blob data will be reformatted
+            according to that profile. This value can be a CSVDialect or a DelimitedJSON.
+        :paramtype output_format: ~azure.storage.blob.CSVDialect or ~azure.storage.blob.DelimitedJSON
+        :keyword bool has_header:
```

Comment:
can we merge has_header into blob_format?

Reply:
I would rather not - this property doesn't make sense when the same object is used for `output_format` and it breaks the csv.Dialect API

Reply:
I'm good with it just it's not consistent with other platforms, while it's a pythonic thing so I guess it's good. @kasobol-msft any thought?

Reply:
I find "CSVDialect" as something to describe generic delimited text a confusing naming (e.g. TSV and CSV are both subclasses of delimited text if we just think about categorizing these formats in general). But I can understand that Python community might be used to it.

As for placing the "has_header" outside I think it's not very cohesive - we pack all csv related attributes into separate class... but one. What if we introduce more CSV related knobs in the future? 
Can't we just extend from CSVDialect and default extra knobs to something reasonable if not specified? Otherwise I'd rather not couple us to CSVDialect and make room for our feature evolution.

Reply:
Thanks - I can call this DelimitedTextDialect if we wish to be generic. Would that be preferable?

Okay - I will move has_header back to the object and use defaults to allow this object to still be interchangable with a Python Dialect.

Reply:
Sounds good.